trunk/src/form/form-edit.js

```diff
@@ -0,0 +1,123 @@
+                dropdownItems.push({form_id: parsedShortcode[0][2], title: parsedShortcode[1][2]})
+            }
+
+            if (shortcodes.length > 0) {
+                this.setState({
+                    dropdownItems: dropdownItems,
+                    showDropdown: true,
+                });
+            }
+
+        })
+
+        xhr.open('POST', ajaxurl)
+        let formData = new FormData();
+        formData.append("action", "get_pardot_forms_shortcode_select_html");
+        xhr.send(formData);
+    }
+
+    // We need getDerivedStateFromProps() and hideOverlay() to allow the user to click anywhere on the block to select it.
+    // I believe the iframe causes issues with the way the Block Editor interacts with user clicks.
+    // We essentially need to create an overlay to capture the first click, then switch to interactive mode,
+    // which allows the user to interact with the iframe.
+    // Source: https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/embed/embed-preview.js#L34
+    static getDerivedStateFromProps(nextProps, state) {
+        if (!nextProps.isSelected && state.interactive) {
+            return {interactive: false};
+        }
+
+        return null;
+    }
+
+    hideOverlay() {
+        this.setState({interactive: true});
+    }
+
+    handleValueOnChange(e) {
+        this.props.setAttributes({
+            [e.target.id]: e.target.value
+        });
+    }
+
+    handleDropdownChange(e) {
+        let index = e.target.selectedIndex;
+        let title = e.target[index].text
+        this.props.setAttributes({form_id: e.target.value, title: title});
+    };
+
+
+    render() {
+        return (
+            <>
+                <InspectorControls>
+                    <div className="sidebar">
+
+                        <h2>{__('Form')}</h2>
+                        {this.state.showDropdown ? <select name="formSelect" value={this.props.attributes.form_id}
+                                                           onChange={e => this.handleDropdownChange(e)}>
+                            <option value={-1} key={-1} label={"Select"}>Select</option>
+                            {this.state.dropdownItems.map(function (n) {
+                                return (<option label={n.title} value={n.form_id} key={n.form_id}>{n.title}</option>);
+                            })}
+                        </select> : <p>It doesn't appear that you have any forms.</p>}
+
+                        <h2>{__('Width')}</h2>
+                        <input type="text" className="textEntry" id="width"
+                               value={this.props.attributes.width}
+                               onChange={(e) => this.handleValueOnChange(e)}
+                               placeholder={__('Must be an integer (e.g. 250)')}/>
+
+                        <h2>{__('Height')}</h2>
+                        <input type="text" className="textEntry" id="height"
```

Comment:
wouldn't it make more sense for these fields to be numeric

also is there validation to ensure that these are ints?

Reply:
I did some inspection that the params eventually become string value in iframe. Like `width="200" height="200"` . We seems never have validation. If I pass non int value in iframe, it seems handles width and height to default 300 * 150, which doesn't break the page.

Reply:
can we easily add some validation here for those non-int values?

Reply:
Actually I realize the reason allowing string was because we allow percentage wise as width and height, such as "80%"